vertical-pod-autoscaler/e2e/updater.go

```diff
@@ -78,20 +63,110 @@ var _ = UpdaterE2eDescribe("Updater", func() {
 				UpperBound:    resourceList,
 			}},
 		}
+		InstallVPA(f, vpaCRD)
 
-		vpaClientSet := vpa_clientset.NewForConfigOrDie(config)
-		vpaClient := vpaClientSet.PocV1alpha1()
-		_, err = vpaClient.VerticalPodAutoscalers(ns).Create(vpaCRD)
+		ginkgo.By("Waiting for pods to be restarted")
+
+		err := waitForPodSetChangedInDeployment(c, d, podList)
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
-		ginkgo.By("Waiting for pods to be restarted")
+	})
 
-		err = waitForPodSetChangedInDeployment(c, d, podList)
+	ginkgo.It("observes pod disruption budget", func() {
+
+		ginkgo.By("Setting up a hamster deployment")
+		c := f.ClientSet
+		ns := f.Namespace.Name
+
+		d, podList := setupHamsterDeployment(f, "10m", "10Mi", 10)
+		podSet := makePodSet(podList)
+
+		ginkgo.By("Setting up prohibitive PDB for hamster deployment")
+		pdb := setupPDB(f, "hamster-pdb", 0 /* maxUnavailable */)
+
+		ginkgo.By("Setting up a VPA CRD")
+		setupVPA(f, "25m")
+
+		ginkgo.By(fmt.Sprintf("Waiting for pods to be restarted, hoping it won't happen, sleep for %s", VpaEvictionTimeout.String()))
+		time.Sleep(VpaEvictionTimeout)
+		restarted, err := isPodSetChanged(c, d, podSet)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(restarted).To(gomega.BeFalse())
+
+		ginkgo.By("Updating the PDB to allow for multiple pods to be evicted")
+		// Creating new PDB and removing old one, since PDBs are immutable at the moment
+		permissiveMaxUnavailable := 7
+		setupPDB(f, "hamster-pdb-2", permissiveMaxUnavailable)
+		err = c.PolicyV1beta1().PodDisruptionBudgets(ns).Delete(pdb.Name, &metav1.DeleteOptions{})
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
+		ginkgo.By(fmt.Sprintf("Waiting for pods to be restarted, sleep for %s", VpaEvictionTimeout.String()))
+		time.Sleep(VpaEvictionTimeout)
+		ginkgo.By("Checking enough pods were restarted.")
+		restartedCount, err := getPodSetChanges(c, d, podSet)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(restartedCount >= permissiveMaxUnavailable).To(gomega.BeTrue())
 	})
 })
 
+func setupHamsterDeployment(f *framework.Framework, cpu, memory string, replicas int32) (*appsv1.Deployment, *apiv1.PodList) {
+	cpuQuantity := ParseQuantityOrDie(cpu)
+	memoryQuantity := ParseQuantityOrDie(memory)
+
+	d := NewHamsterDeploymentWithResources(f, cpuQuantity, memoryQuantity)
+	if replicas > 0 {
```

Comment:
This looks like something that could be improved - either by allowing to pass nil explicitly (I understand it's not too nice for the callers) or at least specifying this behaviour in function comment.

Reply:
Went for passing nil, it's a bit awkward for the caller not not too bad.